src/tools/illink/src/ILLink.Tasks/LinkTask.cs

```diff
@@ -261,9 +261,10 @@ public class ILLink : ToolTask
 				if (!String.IsNullOrEmpty (_illinkPath))
 					return _illinkPath;
 
-				var taskDirectory = Path.GetDirectoryName (Assembly.GetExecutingAssembly ().Location);
+				Version version = Environment.Version;
 				// The linker always runs on .NET Core, even when using desktop MSBuild to host ILLink.Tasks.
-				_illinkPath = Path.Combine (Path.GetDirectoryName (taskDirectory), "net7.0", "illink.dll");
+				_illinkPath = Path.Join (AppDomain.CurrentDomain.BaseDirectory, $"net{version.Major}.{version.Minor}", "illink.dll");
```

Comment:
This doesn't look right. I think `AppDomain.CurrentDomain.BaseDirectory` is the location of the current process, not the task DLL. My understanding of this code is that the task wants to find the location of illink.dll relative to itself. I think the Assembly.Location use was probably correct here, and the suppression is probably appropriate too as we can be confident that MSBuild will never be a single-file app.

Reply:
Debugging using the TestErrorHandling test these are the values: 

Assembly.GetExecutingAssembly ().Location
"runtime\\artifacts\\bin\\ILLink.Tasks.Tests\\Debug\\net7.0\\ILLink.Tasks.dll"
AppDomain.CurrentDomain.BaseDirectory
"runtime\\artifacts\\bin\\ILLink.Tasks.Tests\\Debug\\net7.0\\"
_illinkPath
"runtime\\artifacts\\bin\\ILLink.Tasks.Tests\\Debug\\net7.0\\illink.dll"

So the only part that would be incorrect is adding the "net7.0" folder because it will already be included from getting either Assembly.Location() or BaseDirectory.

Reply:
I talked with Sven about the BaseDirectory location when not executed in the test environment. He agrees that the process location might variate and therefore the right approach is to use the Assembly.Location() and the suppression.